python_utils_test.py

```diff
@@ -0,0 +1,208 @@
+
+        with python_utils.open_file('temp_file.txt', 'rb', encoding=None) as f:
+            content = f.read()
+
+        self.assertIn('<title>Google</title>', content)
+        tmp_file.close()
+
+    def test_url_open(self):
+        response = python_utils.url_open('http://www.google.com')
+        self.assertEqual(response.getcode(), 200)
+        self.assertEqual(response.url, 'http://www.google.com')
+
+    def test_url_request(self):
+        response = python_utils.url_request('http://www.google.com', None, {})
+        self.assertEqual(response.get_full_url(), 'http://www.google.com')
+
+    def test_url_unquote_plus(self):
+        self.assertEqual(
+            python_utils.url_unquote_plus('/El+Ni%C3%B1o/'), '/El Niño/')
+
+    def test_divide(self):
+        self.assertEqual(python_utils.divide(4, 2), 2)
+        self.assertEqual(python_utils.divide(5, 2), 2)
+
+    def test_with_metaclass(self):
+        class BaseForm(python_utils.OBJECT):
+            """Test baseclass."""
+            pass
+
+        class FormType1(type):
+            """Test metaclass."""
+            pass
+
+        class FormType2(type):
+            """Test metaclass."""
+            pass
+
+        class Form(python_utils.with_metaclass(FormType1, BaseForm)): # pylint: disable=inherit-non-class
+            """Test class."""
+            pass
+
+        self.assertTrue(isinstance(Form, FormType1))
+        self.assertFalse(isinstance(Form, FormType2))
+
+    def test_convert_to_bytes(self):
+        string1 = 'Home'
+        string2 = u'Лорем'
+        self.assertEqual(python_utils.convert_to_bytes(string1), string1)
+        self.assertEqual(
+            python_utils.convert_to_bytes(string2),
+            string2.encode(encoding='utf-8'))
+
+    def test_url_split(self):
+        response = python_utils.url_split('http://www.google.com')
+        self.assertEqual(response.geturl(), 'http://www.google.com')
+
+    def test_url_unsplit(self):
+        response = python_utils.url_split('http://www.google.com')
+        self.assertEqual(
+            python_utils.url_unsplit(response), 'http://www.google.com')
+
+    def test_parse_query_string(self):
+        response = python_utils.parse_query_string(
+            'http://www.google.com?search=oppia')
+        self.assertEqual(response, {'http://www.google.com?search': ['oppia']})
+
+    def test_un_quote(self):
+        response = python_utils.un_quote('/El%20Ni%C3%B1o/')
+        self.assertEqual(response, '/El Niño/')
+
+    def test_url_parse(self):
+        response = python_utils.url_parse('http://www.google.com')
+        self.assertEqual(response.geturl(), 'http://www.google.com')
+
+    def test_url_join(self):
+        response = python_utils.url_join(
+            'http://www.cwi.nl/%7Eguido/Python.html', 'FAQ.html')
+        self.assertEqual(response, 'http://www.cwi.nl/%7Eguido/FAQ.html')
+
+
+class PythonUtilsForPython2Tests(test_utils.GenericTestBase):
```

Comment:
Try using:

```python
@unittest.skipUnless(sys.version[0] == '2', 'Test cases for ensuring Python 2 behavior only')
```

And remove all of the conditions. Same for Python 3.

Reply:
Done